wazo_calld/plugins/transfers/state.py

```diff
@@ -220,59 +308,114 @@ def _start(self, context, exten, variables, timeout):
         )
+
+    def _unset_initiator_variables(self):
         ari_helpers.unset_variable(
             self._ari, self._amid, self.transfer.initiator_call, 'XIVO_TRANSFER_ID'
         )
         ari_helpers.unset_variable(
             self._ari, self._amid, self.transfer.initiator_call, 'XIVO_TRANSFER_ROLE'
         )
 
-        if self.transfer.transferred_call:
-            try:
-                self._ari.channels.hangup(channelId=self.transfer.transferred_call)
-            except ARINotFound:
-                pass
-
-        self._notifier.abandoned(self.transfer)
+    def _set_transferred_variables(self):
+        self._ari.channels.setChannelVar(
+            channelId=self.transfer.transferred_call,
+            variable='XIVO_TRANSFER_ROLE',
+            value='transferred',
+        )
+        self._ari.channels.setChannelVar(
+            channelId=self.transfer.transferred_call,
+            variable='XIVO_TRANSFER_ID',
+            value=self.transfer.id,
+        )
 
-    def _cancel(self):
+    def _unset_transferred_variables(self):
         ari_helpers.unset_variable(
             self._ari, self._amid, self.transfer.transferred_call, 'XIVO_TRANSFER_ID'
         )
         ari_helpers.unset_variable(
             self._ari, self._amid, self.transfer.transferred_call, 'XIVO_TRANSFER_ROLE'
         )
+
+    def _set_recipient_variables(self):
+        self._ari.channels.setChannelVar(
+            channelId=self.transfer.recipient_call,
+            variable='XIVO_TRANSFER_ROLE',
+            value='recipient',
+        )
+        self._ari.channels.setChannelVar(
+            channelId=self.transfer.recipient_call,
+            variable='XIVO_TRANSFER_ID',
+            value=self.transfer.id,
+        )
+
+    def _unset_recipient_variables(self):
         ari_helpers.unset_variable(
-            self._ari, self._amid, self.transfer.initiator_call, 'XIVO_TRANSFER_ID'
+            self._ari, self._amid, self.transfer.recipient_call, 'XIVO_TRANSFER_ID'
         )
         ari_helpers.unset_variable(
-            self._ari, self._amid, self.transfer.initiator_call, 'XIVO_TRANSFER_ROLE'
+            self._ari, self._amid, self.transfer.recipient_call, 'XIVO_TRANSFER_ROLE'
         )
 
+    def _abandon(self):
+        # transferred channel left transfer
+
+        if self.transfer.transferred_call:
+            try:
+                self._ari.channels.hangup(channelId=self.transfer.transferred_call)
+            except ARINotFound:
+                pass
+
+        self._notifier.abandoned(self.transfer)
+
+    def _cancel(self):
+        # transfer is cancelled: transfer recipient exits
+        # initiator and transferred channels remain
+        # cancel logic:
+        # 1. unset transfer channel variables
+        # 2. hangup transfer recipient channel
+        # 3. stop ringing on initiator channel
+        # 4. remove transferred party from hold
+        # 5. bridge back transferred and initiator parties
```

Comment:
The comment about cancel logic seems redundant to me, and risks becoming obsolete at the next change in this function. I suggest removing it, or extracting more logic to functions to make it more readable.